homeassistant/components/ping/binary_sensor.py

```diff
@@ -72,20 +84,28 @@ async def async_setup_platform(
         ping_cls = PingDataICMPLib
 
     async_add_entities(
-        [PingBinarySensor(name, ping_cls(hass, host, count, privileged))]
+        [PingBinarySensor(name, entry, ping_cls(hass, host, count, privileged))]
     )
 
 
 class PingBinarySensor(RestoreEntity, BinarySensorEntity):
     """Representation of a Ping Binary sensor."""
 
     _attr_device_class = BinarySensorDeviceClass.CONNECTIVITY
-
-    def __init__(self, name: str, ping: PingDataSubProcess | PingDataICMPLib) -> None:
+    _attr_available = False
+
+    def __init__(
+        self,
+        name: str,
+        config_entry: ConfigEntry,
+        ping: PingDataSubProcess | PingDataICMPLib,
+    ) -> None:
         """Initialize the Ping Binary sensor."""
-        self._attr_available = False
         self._attr_name = name
+        self._attr_unique_id = f"{config_entry.entry_id}_binary_sensor"
```

Comment:
```suggestion
        self._attr_unique_id = config_entry.entry_id
```

unique ids are per domain/platform so you can drop `binary_sensor` 

https://github.com/home-assistant/core/blob/82ce5e56b5e610ced30838e805b0bf7b49a1740e/homeassistant/components/homekit/aidmanager.py#L36